bundles/org.openhab.binding.tapocontrol/src/main/java/org/openhab/binding/tapocontrol/internal/api/protocol/klap/KlapCipher.java

```diff
@@ -0,0 +1,203 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.tapocontrol.internal.api.protocol.klap;
+
+import static org.openhab.binding.tapocontrol.internal.constants.TapoErrorCode.*;
+import static org.openhab.binding.tapocontrol.internal.helpers.TapoEncoder.*;
+import static org.openhab.binding.tapocontrol.internal.helpers.TapoUtils.*;
+
+import java.math.BigInteger;
+import java.security.NoSuchAlgorithmException;
+
+import javax.crypto.Cipher;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.tapocontrol.internal.helpers.TapoErrorHandler;
+
+/**
+ * Cipher for KLAP-Protocol
+ *
+ * @author Christian Wild - Initial contribution
+ */
+@NonNullByDefault
+public class KlapCipher {
+    private int ivSeq = 0;
+    private byte[] iv;
+    private byte[] key;
+    private byte[] sig;
+
+    /* class contants */
```

Comment:
Maybe just cosmetic but I like to see static final at the very beginning of the class.